File: app/src/main/java/com/mohsenoid/rickandmorty/view/character/details/CharacterDetailsContract.java
package com.mohsenoid.rickandmorty.view.character.details;

import com.mohsenoid.rickandmorty.domain.entity.CharacterEntity;
import com.mohsenoid.rickandmorty.view.base.BasePresenter;
import com.mohsenoid.rickandmorty.view.base.BaseView;

public interface CharacterDetailsContract {

    interface View extends BaseView {

        void onNoOfflineData();

        void setCharacter(CharacterEntity character);
    }

    interface Presenter extends BasePresenter<View> {

        void setCharacterId(int characterId);

        void loadCharacter();
    }
}

File: app/src/main/kotlin/com/mohsenoid/rickandmorty/view/character/details/CharacterDetailsContract.kt
package com.mohsenoid.rickandmorty.view.character.details

import com.mohsenoid.rickandmorty.domain.model.ModelCharacter
import com.mohsenoid.rickandmorty.view.base.BasePresenter
import com.mohsenoid.rickandmorty.view.base.BaseView

interface CharacterDetailsContract {

    interface View : BaseView {

        fun onNoOfflineData()

        fun setCharacter(character: ModelCharacter)
    }

    abstract class Presenter : BasePresenter<View>() {

        abstract suspend fun loadCharacter(characterId: Int)
    }
}
